app/qml/account/MMHowYouFoundUsPage.qml

```diff
@@ -24,6 +24,38 @@ MMPage {
 
   signal howYouFoundUsSelected( var selectedText )
 
+  ListModel {
+    id: sourceListModel
+
+    Component.onCompleted: {
+      var items = [
+        { name: qsTr( "Search engine (Google, ...)" ), key: "search_engine", icon: __style.searchIcon, submenu: false },
+        { name: qsTr( "Blog" ), key: "blog", icon: __style.termsIcon, submenu: false },
+        { name: qsTr( "Word of Mouth" ), key: "mouth", icon: __style.mouthIcon, submenu: false },
+        { name: qsTr( "QGIS website" ), key: "qgis_website", icon: __style.qgisIcon, submenu: false },
+        { name: qsTr( "Application store" ), key: "app_store", icon: __style.subscriptionsIcon, submenu: false },
+        { name: qsTr( "Teacher" ), key: "teacher", icon: __style.teacherIcon, submenu: false },
+        { name: qsTr( "Conference" ), key: "conference", icon: __style.briefcaseIcon, submenu: false },
+        { name: qsTr( "Social media" ), key: "social", icon: __style.socialMediaIcon, submenu: false }
+      ];
+      var otherItem = { name: qsTr( "Other" ), key: "other", icon: __style.otherIcon, submenu: false };
+
+      for ( var i = items.length - 1; i > 0 ; i-- ) {
+        var j = Math.floor( Math.random() * ( i + 1 ) );
+        var temp = items[ i ];
+        items[ i ] = items[ j ];
```

Comment:
I can see it is motivated by this [algo](https://stackoverflow.com/a/12646864/7875594) which if fine, but I struggle to understand how come this won't lead to exceptions.
If you consider this scenario:

```
array length = 8

for loop ( i == 7 )
  j = Math.floor( 0.999 * ( 7 + 1 ) ) // j == 8 here
  ..
  items [j] // <<-- here you go outside of bounds of the input array
```

For that reason, I would prefer a simpler and more readable version: https://stackoverflow.com/a/46545530/7875594 :)